modules/dynamic/src/smithy4s/dynamic/DynamicService.scala

```diff
@@ -29,7 +29,7 @@ case class DynamicService(
       op: DynamicOp[I, E, O, SI, SO]
   ): (I, Endpoint[DynamicOp, I, E, O, SI, SO]) = {
     val endpoint = endpointMap
-      .get(op.id)
+      .getOrElse(op.id, sys.error("Unknown endpoint: " + op.id))
```

Comment:
this was a fun one

Reply:
Yeah, I wish there was some kind-polymorphic map implementation. That'd solve a lot of problems.